### DISCUSSION TITLE
How to get selected files count from QUploader?

### DISCUSSION THREAD

**Discussion by oneriang** [quasarframework/quasar#14551] [General - Components / Directives / etc] [answered]
How to get selected files count from QUploader?
I try to use property '.files' of $refs, but it is allways 0.

<img width="426" alt="image" src="https://user-images.githubusercontent.com/5113931/193847319-b49bc785-27c0-4145-8463-0d44f36c23a3.png">

**Comment by Diferno** [comment, marked as answer]
I use the added event to get the files that have been added. https://v1.quasar.dev/vue-components/uploader#quploader-api
```javascript
@added -> function(files)
  Description
  Emitted when files are added into the list
  Parameters
  files : Array
  Description
  Array of files that were added`

**Reply by oneriang** [reply]
Thanks.
I have one more question.
When added, save to the 'selected files' array.
When delete a file, also need to determine which file is being deleted.
That said, I also need to maintain this 'selected files' array.
So I hope property '.files' of $refs works well

**Reply by metalsadman** [reply]
Please convert this discussion into a bug report like i asked you to in discord.
